### DISCUSSION TITLE
Unable to correlate spans with tracing-opentelemetry

### DISCUSSION THREAD

**Discussion by davidbarsky** [tokio-rs/tracing#1991] [Q&A] [answered]
Hi folks, I found this old issue and I'm trying to get `trucing-opentelemetry` and Jaeger to work together.    I am able to get spans into Jaeger, however I cannot find a way to tie the spans together with trace propagation.  I'm trying to use code like the following:

```rust
let context = global::get_text_map_propagator(|propagator| propagator.extract(&carrier))
span.set_parent(context);
```

Does it matter what is in the hashmap in carrier?

Thanks.

_Originally posted by @velvia in https://github.com/tokio-rs/tracing/issues/89#issuecomment-1065689061_

**Comment by davidbarsky** [comment, marked as answer]
> Does it matter what is in the hashmap in carrier?

@velvia: Yes. The HashMap in the carrier should have the trace ID that you're looking to correlate. Can you post more of your code? How closely are you following [this documentation](https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/trait.OpenTelemetrySpanExt.html#tymethod.set_parent)?

**Reply by velvia** [reply]
@davidbarsky I'm following the example pretty closely.   Actually the example does not have anything in the hash map, which is the critical part.   My code is like this:

```
    let mut carrier = HashMap::new();
    carrier.insert("traceparent".to_string(), hex::encode(digest.0));
    global::get_text_map_propagator(|propagator| propagator.extract(&carrier))
```

You are probably wondering what digest is.   Does the trace parent really need to be a preexisting trace ID, or could it be anything used to tie different spans together?
Are there any other fields that are needed?

Thanks.
